aten/src/ATen/core/type.cpp

```diff
@@ -585,6 +585,22 @@ bool is_null_or_equal(c10::optional<T> a, c10::IntArrayRef b) {
   return !a.has_value() || a.value() == b;
 }
 
+
+
+ bool matchStrides(TensorType* ttp, const at::Tensor& t) {
+   TORCH_INTERNAL_ASSERT(t.is_contiguous());
+   auto ssize = *ttp->stride_properties().size();
```

Comment:
there's an implicit pre-condition though. `matchTensor` is only called by a `TypeCheck` which we only insert on full types for the foreseeable future at least. Maybe it will make sense to move matchTensor to `interpreter.cpp` closer to the only caller?

Reply:
We have another PR that is accepted that does exactly what you're describing; inserts a typecheck without stride information (T-vi's autograd PR). I think this precondition is really far from the source. We should consider a  way to refactor this. Potentially by emitting different op-codes for TypeCheck depending on on the type we're  checking against; i'm not sure.

Reply:
ah right. I think using different op-codes make a lot of sense: CheckGrad, CheckCompleteType, and GenericTypeCheck if we need. I think even with the current TensorType structures we can do checks relatively efficiently if we stop using checks that build intermediate structures and just directly check for the things needed.